indolaw-parser/parser_utils.py

```diff
@@ -847,29 +847,48 @@ def clean_squashed_page_numbers(law: List[str]) -> List[str]:
     print_section_header('CLEANING SQUASHED PAGE NUMBER...')
 
     new_law = []
-    for idx, line in enumerate(law):
+    
+    idx = 0
+    last_idx = len(law)
+    last_input_idx = 0
+
+    while idx < last_idx:
```

Comment:
i think it's clearer if you just do `while idx < len(law)` since you only ever use the `last_idx` variable once anyways

Reply:
If I use `while idx < len(law)` wouldn't that mean the python recalculates `len(law)` for every loop?

Reply:
good point; pulling it out into a variable is fine then

(altho, technically len(law) != last_idx semantically, since the length of an array is technically 1 larger than the last index of the array due to 0-based indexing)